scripts/run_social_messaging.py

```diff
@@ -0,0 +1,149 @@
+from pocs.utils.messaging import PanMessaging
+from pocs.utils.social_twitter import SocialTwitter
+from pocs.utils.social_slack import SocialSlack
+
+the_root_logger = None
+
+
+def say(fmt, *args, error=False):
+    if args:
+        msg = fmt.format(*args)
+    else:
+        msg = fmt
+    if error:
+        print(msg, file=sys.stderr)
+        the_root_logger.error(msg)
+    else:
+        print(msg)
+        the_root_logger.info(msg)
+
+
+def check_social_messages_loop(msg_port, social_twitter, social_slack):
+    cmd_social_subscriber = PanMessaging.create_subscriber(msg_port, 'PANCHAT')
+
+    poller = zmq.Poller()
+    poller.register(cmd_social_subscriber.socket, zmq.POLLIN)
+
+    try:
+        while True:
+            # Poll for messages
+            sockets = dict(poller.poll(500))  # 500 ms timeout
+
+            if cmd_social_subscriber.socket in sockets and \
+                    sockets[cmd_social_subscriber.socket] == zmq.POLLIN:
+
+                msg_type, msg_obj = cmd_social_subscriber.receive_message(flags=zmq.NOBLOCK)
+
+                # Check the various social sinks
+                if social_twitter is not None:
+                    social_twitter.send_message(msg_obj['message'], msg_obj['timestamp'])
+
+                if social_slack is not None:
+                    social_slack.send_message(msg_obj['message'], msg_obj['timestamp'])
+
+            time.sleep(1)
+    except KeyboardInterrupt:
+        pass
+
+
+def run_social_sinks(msg_port, social_twitter, social_slack):
+    the_root_logger.info('Creating sockets')
+
+    threads = []
+    
+    name='social_messaging'
+
+    t = threading.Thread(
+            target=check_social_messages_loop, name=name, args=(msg_port, social_twitter, social_slack), daemon=True)
+    the_root_logger.info('Starting thread {}', name)
+    t.start()
+    threads.append(t)
+
+    time.sleep(0.05)
+    if not any([t.is_alive() for t in threads]):
+        say('Failed to start social sinks thread!', error=True)
+        sys.exit(1)
+    else:
+        the_root_logger.info('Started social messaging')
+        print()
+        print('Hit Ctrl-c to stop')
+    try:
+        # Keep running until they've all died.
+        while threads:
+            for t in threads:
+                t.join(timeout=100)
+                if t.is_alive():
+                    continue
+                say('Thread {} has stopped', t.name, error=True)
+                threads.remove(t)
+                break
+        # If we get here, then the forwarders died for some reason.
+        sys.exit(1)
+    except KeyboardInterrupt:
+        sys.exit(0)
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(
+        description='Run social messaging to forward platform messages to social channels.')
+    parser.add_argument(
+        '--from_config',
+        action='store_true',
+        help='Read social channels config from the pocs.yaml and pocs_local.yaml config files.')
```

Comment:
At this point we don't really have any need for argparse, right? If you agree, I'd drop it until needed.

Reply:
@jamessynge you mean because there is only one option or just because it is the default?

I'd prefer that if we parse _any_ command line options we do it with ArgParse, even if it is only one argument. It's more consistent and ArgParse looks cleaner than using `sys.argv`.

Reply:
However, what we haven't done is establish conventions on naming arguments from the command line. I think the standard would be to always do hypens instead of underscores on the command line, which ArgParse automatically adjusts. So `--from-config` on the command line becomes `args.from_config`.

Reply:
So @wtgee, @jamessynge  nothing to do here then right? BTW I lifted this code from `run_messaging_hub.py`